### PR TITLE
Render prop picker improvements

### PR DESCRIPTION
**Pull request by Rheeseyb**
Follow on from #5101 

This PR introduces the following tweaks to the Component Picker popup that is used for render prop slots:

1. I've split up the `ComponentPicker` component into several components. _For now_ I have kept them all in `component-picker.tsx`, as their individual locations (and that file's location) would have to change anyway when re-using this for the Insert Menu pane, so I'd rather wait until then before further generalising it.
2. We now use the `pointer` cursor when hovering over the clickable elements in the popup
3. We now also use the `pointer` cursor when hovering over the slot itself in the navigator, meaning that it is now clear you can click the slot for a render prop (note also that you now click the empty slot itself rather than the actual row, as per the gif below)
4. All registered components now come with an `(empty)` variant, which just inserts the component as `<MyComponent />`

[**Try it here**](https://utopia.fish/p/2e3389de-emphasized-trumpet/?branch_name=feature-render-prop-insert-picker-2)

component-picker

### CODE REVIEW

**Review by bkrmendy — APPROVED** [review]

File: editor/src/components/navigator/navigator-item/component-picker.tsx [hunk 1]
```diff
@@ -76,143 +63,259 @@ export const ComponentPicker = React.memo((props: ComponentPickerProps) => {
...
+          isSelected={selectedTab === 'preferred'}
+          onClick={switchToPreferredTab}
+        />
+        <PickerTabButton
+          title={'All Components'}
+          isSelected={selectedTab === 'all'}
+          onClick={switchToAllTab}
+        />
+        <div style={{ flexGrow: 1 }} />
+        {onClickCloseButton != null ? (
```
[the remark this answers is not in the document]
Comment (Rheeseyb) [reply editor/src/components/navigator/navigator-item/component-picker.tsx:149]: I originally went with that, but annoyingly it prevents TS from inferring that `onClickCloseButton` is non-null

[the remark this answers is not in the document]
Comment (Rheeseyb) [reply editor/src/components/navigator/navigator-item/component-picker.tsx:149]: Oh wait, that actually doesn't matter anymore, so I _can_ replace it with a `when`!

[the remark this answers is not in the document]
Comment (Rheeseyb) [reply editor/src/components/navigator/navigator-item/component-picker.tsx]: Done 👍

[the remark this answers is not in the document]
Comment (ruggi) [reply editor/src/components/navigator/navigator-item/component-picker.tsx]: 🎖️

**Pull request merged**